templates/sectionsummary.html

```diff
@@ -26,8 +26,10 @@ <h1>{{content.summary.title}}</h1>
     <h1>{{content.summary.title}}</h1>
     {% for summary in content.summary.custom_summary %}
       {% if summary.type == 'List' %}
-
-          <div {{'class="u-mb-l"' if not loop.last }} data-qa="{{summary.list_name}}-list-summary">
+          {% set add_link = summary.add_link %}
+          {% set add_link_text = summary.add_link_text %}
+          {% set list_name = summary.list_name %}
+          <div {{'class="u-mb-l"' if not loop.last }} data-qa="{{list_name}}-list-summary">
           <h2>{{summary.title}}</h2>
           {% if summary.list and summary.list.list_items %}
```

Comment:
Now that the summary component support everything we need for list collectors, I don't think any of the logic around list items are needed. The component support, list items, add links, empty placeholder text as well as the group header. So it can be updated to something like:
```jinja2
        {% if summary.type == 'List' %}
          {% set add_link = summary.add_link %}
          {% set add_link_text = summary.add_link_text %}
          {% set empty_list_text = summary.empty_list_text %}
          {% set summary_title = summary.title %}

          <div {{'class="u-mb-l"' if not loop.last }} data-qa="{{summary.list_name}}-list-summary">
          {% if summary.list %}
            {% set list = summary.list %}
            {% include 'partials/summary/list-summary.html' %}
          {% endif %}
          </div>
        {% endif %}
```

Then `list-summary.html` just needs updating to something like:
```jinja2
{% from "components/summary/_macro.njk" import onsSummary %}

  {% if list.editable %}
    {% set headers = ["Name of person", "Action"] %}
    {% set rows = map_list_collector_config(
      list.list_items,
      "person",
      _("Change"),
      _("Change details for {item_name}"),
      _("Remove"),
      _("Remove {item_name}")
    ) %}
  {% else %}
    {% set headers = ["Name of person"] %}
    {% set rows = map_list_collector_config(
      list.list_items,
      "person"
    ) %}
  {% endif %}

  {% set groups_config = {
    "groupTitle": summary_title,
    "headers": headers,
    "rows": rows,
    "summaryLink": {"url": add_link, "text": add_link_text, "dataQa": "add-item-link"},
  } %}

  {% if empty_list_text %}
    {% do groups_config.update({"placeholderText": empty_list_text}) %}
  {% endif %}

  {{ onsSummary({
    "withinQuestion": true,
    "summaries": [
      {
        "groups": [
          groups_config
        ]
      }
    ],
    "classes": "u-mt-s u-mb-m"
  }) }}
```

This means there is one template for list summary and it is using the actual DS component as intended.

With the proposed changes, the padding of the continue button is slightly large and may need reducing.

Reply:
@MebinAbraham I used your code from above, extra space on the summary was caused by `u-mb-m` class used in our onsSummary I believe. So I removed that class and spacing is the same as in staging now.